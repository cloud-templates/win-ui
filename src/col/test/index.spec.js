import Col from '..';
import Row from '../../row';
import { mount } from '../../../test';

test('Col click event', () => {
  const wrapper = mount(Col);
  wrapper.trigger('click');

  expect(wrapper.emitted('click')).toBeTruthy();
});

test('Row click event', () => {
  const wrapper = mount(Row);
  wrapper.trigger('click');

  expect(wrapper.emitted('click')).toBeTruthy();
});

test('gutter prop', () => {
  const wrapper = mount({
    template: `
      <win-row gutter="24">
        <win-col span="24">24</win-col>

        <win-col span="12">12</win-col>
        <win-col span="12">12</win-col>

        <win-col span="8">8</win-col>
        <win-col span="8">8</win-col>
        <win-col span="8">8</win-col>

        <win-col span="6">6</win-col>
        <win-col span="6">6</win-col>
        <win-col span="6">6</win-col>
        <win-col span="6">6</win-col>

        <win-col span="7">7</win-col>
        <win-col span="6">6</win-col>
        <win-col span="5">5</win-col>
        <win-col span="4">4</win-col>
        <win-col span="3">3</win-col>
        <win-col span="2">2</win-col>
      </win-row>
    `,
  });

  expect(wrapper).toMatchSnapshot();
});
