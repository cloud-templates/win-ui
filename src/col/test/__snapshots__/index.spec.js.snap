// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gutter prop 1`] = `
<div class="win-row">
  <div class="win-col win-col--24">24</div>
  <div class="win-col win-col--12" style="padding-right: 12px;">12</div>
  <div class="win-col win-col--12" style="padding-left: 12px;">12</div>
  <div class="win-col win-col--8" style="padding-right: 16px;">8</div>
  <div class="win-col win-col--8" style="padding-left: 8px; padding-right: 8px;">8</div>
  <div class="win-col win-col--8" style="padding-left: 16px;">8</div>
  <div class="win-col win-col--6" style="padding-right: 18px;">6</div>
  <div class="win-col win-col--6" style="padding-left: 6px; padding-right: 12px;">6</div>
  <div class="win-col win-col--6" style="padding-left: 12px; padding-right: 6px;">6</div>
  <div class="win-col win-col--6" style="padding-left: 18px;">6</div>
  <div class="win-col win-col--7" style="padding-right: 18px;">7</div>
  <div class="win-col win-col--6" style="padding-left: 6px; padding-right: 12px;">6</div>
  <div class="win-col win-col--5" style="padding-left: 12px; padding-right: 6px;">5</div>
  <div class="win-col win-col--4" style="padding-left: 18px;">4</div>
  <div class="win-col win-col--3" style="padding-right: 12px;">3</div>
  <div class="win-col win-col--2" style="padding-left: 12px;">2</div>
</div>
`;
